From 18e9bb957b758d4acd82fce1b7eb40af0eaa5b7b Mon Sep 17 00:00:00 2001
From: Jorge Niedbalski <jnr@metaklass.org>
Date: Thu, 7 Nov 2019 18:30:22 -0300
Subject: [PATCH] Add loadbalancer resource type.

Adds the loadbalancer resource type for usage
with the network.services.lb.* metrics.

Depends-On: https://review.opendev.org/#/c/695855/
Change-Id: Ib6e78438c3da0e22d93f720f00cdeadf0ed7a91f
Closes-Bug: #1848286
Signed-off-by: Jorge Niedbalski <jnr@metaklass.org>
---
 ceilometer/gnocchi_client.py                     |  6 ++++++
 ceilometer/publisher/data/gnocchi_resources.yaml | 12 ++++++++++++
 2 files changed, 18 insertions(+)

diff --git a/ceilometer/gnocchi_client.py b/ceilometer/gnocchi_client.py
index e147fd80..cdecae6a 100644
--- a/ceilometer/gnocchi_client.py
+++ b/ceilometer/gnocchi_client.py
@@ -224,6 +224,12 @@ resources_update_operations = [
          {"op": "add", "path": "/attributes/instance_id",
           "value": {"type": "uuid", "required": False}},
      ]},
+    {"desc": "add loadbalancer resource type",
+     "type": "create_resource_type",
+     "resource_type": "loadbalancer",
+     "data": [{
+         "attributes": {}
+     }]},
 ]
 
 # NOTE(sileht): We use LooseVersion because pbr can generate invalid
diff --git a/ceilometer/publisher/data/gnocchi_resources.yaml b/ceilometer/publisher/data/gnocchi_resources.yaml
index 5186c4d8..b1234ded 100644
--- a/ceilometer/publisher/data/gnocchi_resources.yaml
+++ b/ceilometer/publisher/data/gnocchi_resources.yaml
@@ -386,3 +386,15 @@ resources:
     attributes:
       controller: resource_metadata.controller
       switch: resource_metadata.switch
+
+  - resource_type: loadbalancer
+    metrics:
+      network.services.lb.outgoing.bytes:
+      network.services.lb.incoming.bytes:
+      network.services.lb.pool:
+      network.services.lb.listener:
+      network.services.lb.member:
+      network.services.lb.health_monitor:
+      network.services.lb.loadbalancer:
+      network.services.lb.total.connections:
+      network.services.lb.active.connections:
\ No newline at end of file
-- 
2.24.0

